pkg/install/service.go

```diff
@@ -30,14 +30,16 @@ func EnsureService(args []string) error {
 	var k0sDisplayName string
 
 	prg := &program{}
-
 	for _, v := range args {
 		if v == "server" {
 			k0sDisplayName = "k0s server"
+			k0sServiceName = k0sServiceName + v
+		} else if v == "worker" {
```

Comment:
Can this somehow go to the `else` branch? 

worker` doesn't seem to get a `k0sDisplayName` like before.

Looks to me like the whole conditional could be skipped and replaced with:
```go
for _, v := range args {
  k0sDisplayName = "k0s " + v
  k0sServiceName = "k0s" + v
}
```

I have no clue what it is doing, but looks weird.

Reply:
Or maybe inside `if v == "server" || v == "worker"`

Reply:
Display name and service name are not the same. 
Service name is the name of  the service stub like k0sserver.service or k0sworker.service where the display name is just how it is going to appear when listed with systemctl.
p.s. Args slice contains more things than just role name so it had to have the condition as above

Reply:
Probably doesn't matter, but maybe it should break after the first match.

Reply:
Could have been simply
```go
  for _, v := range args {
    if v == "server" || v == "worker" {
      k0sDisplayName = "k0s " + v
      k0sServiceName = k0sServiceName + v
      break
    }
  }
```

Reply:
Fixed it